### PR TITLE
added diskType to support shuffle optimized disk configuration

### PR DESCRIPTION
**Pull request by nithu0115**
*Issue #, if available:* NA

*Description of changes:*
added diskType to support shuffle optimized disk configuration

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by itzmadhan — APPROVED** [review]

File: aws-emrserverless-application/aws-emrserverless-application.json [hunk 1]
```diff
@@ -433,6 +437,11 @@
       "maxLength": 15,
       "pattern": "^[1-9][0-9]*(\\s)?(GB|gb|gB|Gb)$"
     },
+    "DiskType": {
+      "description": "Per worker DiskType resource. Shuffle optimized and Standard are only supported types and specifying diskType is optional",
```
Comment (oleglypkan) [inline aws-emrserverless-application/aws-emrserverless-application.json:441]: Do we want to say that it defaults to Standard when unspecified?

File: aws-emrserverless-application/docs/workerconfiguration.md [hunk 2]
```diff
@@ -72,3 +74,15 @@ _Pattern_: <code>^[1-9][0-9]*(\s)?(GB|gb|gB|Gb)$</code>
...
+
+Per worker DiskType resource. Shuffle optimized and Standard are only supported types and specifying diskType is optional
+
+_Required_: No
+
+_Type_: String
+
+_Pattern_: <code>^(SHUFFLE_OPTIMIZED|[Ss]huffle_[Oo]ptimized|STANDARD|[Ss]tandard)$</code>
+
+_Update requires_: [No interruption](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-update-behaviors.html#update-no-interrupt)
```
Comment (oleglypkan) [inline aws-emrserverless-application/docs/workerconfiguration.md:87]: I am not sure about this. If an application is started, update will fail and require stopping the application to perform an update. Does this use case still meet "No interruption" definition?

File: aws-emrserverless-application/aws-emrserverless-application.json [hunk 3]
```diff
@@ -200,6 +200,10 @@
         "Disk": {
           "description": "Per worker Disk resource. GB is the only supported unit and specifying GB is optional",
           "$ref": "#/definitions/DiskSize"
+        },
+        "DiskType": {
+          "description": "Per worker DiskType resource. Shuffle optimized and Standard are only supported types and specifying diskType is optional",
```
Comment (oleglypkan) [inline aws-emrserverless-application/aws-emrserverless-application.json:205]: Do we want to say that it defaults to Standard when unspecified?

File: aws-emrserverless-application/aws-emrserverless-application.json (hunk 1 above)
Comment (nithu0115) [reply aws-emrserverless-application/aws-emrserverless-application.json:441]: These are generated by model.

File: aws-emrserverless-application/docs/workerconfiguration.md (hunk 2 above)
Comment (nithu0115) [reply aws-emrserverless-application/docs/workerconfiguration.md:87]: This docs are auto-generated. It probably assumes that update is only possible when application is stopped. And in that case, there will be no interruptions.

File: aws-emrserverless-application/aws-emrserverless-application.json (hunk 3 above)
Comment (nithu0115) [reply aws-emrserverless-application/aws-emrserverless-application.json:205]: Same here. auto generated docs based on model

**Review by oleglypkan — APPROVED** [review]

**Comment by vermanish** [comment]
Looks good

**Review by vermanish — APPROVED** [review]
Looks good

**Pull request merged**